### PR TITLE
Configurable nextcloud lifecycle hooks and extra environment

### PR DESCRIPTION
**Pull request by ghost**
#### Is this a new chart

no

#### What this PR does / why we need it:

Make nextcloud chart more configurable and therfore more generally usable.

#### Which issue this PR fixes

none, add functionality

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

File: stable/nextcloud/templates/deployment.yaml [hunk 1]
```diff
@@ -34,6 +34,21 @@ spec:
       - name: {{ .Chart.Name }}
         image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
         imagePullPolicy: {{ .Values.image.pullPolicy }}
+        {{- if .Values.lifecycle }}
```
Comment (billimek) [inline stable/nextcloud/templates/deployment.yaml:37]: Can you please also add documentation for the `lifecycle` value to the `README.md`?

**Review by billimek — CHANGES_REQUESTED** [review]

File: stable/nextcloud/templates/deployment.yaml (hunk 1 above)
Comment (ghost) [reply stable/nextcloud/templates/deployment.yaml:37]: Added and updated the PR.

**Review by billimek — APPROVED** [review]

**Pull request merged**